src/core/reducers/addons.js

```diff
@@ -166,14 +204,7 @@ export function createInternalAddon(apiAddon: ExternalAddonType): AddonType {
     weekly_downloads: apiAddon.weekly_downloads,
 
     // These are custom properties not in the API response.
-
-    platformFiles: {
-      all: undefined,
-      android: undefined,
-      linux: undefined,
-      mac: undefined,
-      windows: undefined,
-    },
+    platformFiles: createPlatformFiles(apiAddon.current_version),
```

Comment:
How about `currentPlatformFiles` ?

However, I don't totally understand why this is needed. Can't the `versions` reducer respond to the same load action and store a version object like any other version object? After that, we could add `addon.currentVersionId` as a link to the version object.

Reply:
I hadn't considered it when writing this patch, but now I'm working on https://github.com/mozilla/addons-frontend/issues/6634 and it's become obvious that something needs to change. Using an id is an interesting idea I hadn't considered. So we'd store the "real" version info in the versions state and just store a link to it in the addon? It feels a bit weird to reference an item in a different reducer's state, but I guess that's ok. Is that what you're suggesting?